rcl/test/memory_tools/memory_tools.hpp

```diff
@@ -104,4 +104,29 @@ RCL_MEMORY_TOOLS_PUBLIC
 void
 memory_checking_thread_init();
 
-#endif  // MEMORY_TOOLS_HPP_
+// What follows is a set of failing allocator functions, used for testing.
+void *
+failing_malloc(size_t size, void * state)
+{
+  (void)(size);
```

Comment:
Should we remove the paranthesis around the variable name? Other existing code (even in this PR https://github.com/ros2/rcl/pull/21/files#diff-8fda46206fc38bfdbcac0e5213bffddaR443) avoids them.

Same a few times below.